### PR TITLE
🌱 Release v1.7.0-beta.1

### PR DESCRIPTION
**Pull request by willie-yao**
/area release

### CODE REVIEW

**Comment by mboersma** [comment]
/retest

Broken link seems to work for me:  https://github.com/go-gl/glfw/compare/v3.3/glfw/6f7a984...v3.3/glfw/12ad95a ?

**Comment by willie-yao** [comment]
@mboersma Ah we saw this in the previous beta release. We decided to remove that line. I'll go ahead and make the change here! https://github.com/kubernetes-sigs/cluster-api/pull/10253#issuecomment-1989514679

File: CHANGELOG/v1.7.0-beta.1.md [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+- 298 new commits merged
+- 4 breaking changes :warning:
+- 11 feature additions ✨
+- 36 bugs fixed 🐛
+
+## :memo: Proposals
+- MachinePool: Update MachinePool Machines proposal with diagrams (#9664)
+
+## :warning: Breaking Changes
+- API: Remove v1alpha4 API Version (#9939)
```
Comment (sbueringer) [inline CHANGELOG/v1.7.0-beta.1.md:30]: This got reverted later in https://github.com/kubernetes-sigs/cluster-api/pull/10147
I think it's confusing if it still ends up here as a breaking change

File: CHANGELOG/v1.7.0-beta.1.md [hunk 2]
```diff
@@ -0,0 +1,411 @@
+🚨 This is a RELEASE CANDIDATE. Use it only for testing purposes. If you find any bugs, file an [issue](https://github.com/kubernetes-sigs/cluster-api/issues/new).
+
+## Highlights
+
+* Enable kubeadm feature gates mutation
```
Comment (sbueringer) [inline CHANGELOG/v1.7.0-beta.1.md:5]: We probably want to write some more here when we get closer to the release (I currently just don't have the time to go through all the changes below to write something up)

Likely candidates are the changes below New Features

**Review by sbueringer — COMMENTED** [review]
Overall okay for me for this release

File: CHANGELOG/v1.7.0-beta.1.md [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+## 👌 Kubernetes version support
+
+- Management Cluster: v1.26.x -> v1.29.x
+- Workload Cluster: v1.24.x -> v1.29.x
+
+[More information about version support can be found here](https://cluster-api.sigs.k8s.io/reference/versions.html)
+
+## Changes since v1.6.0
+## :chart_with_upwards_trend: Overview
+- 298 new commits merged
```
Comment (cahillsf) [inline CHANGELOG/v1.7.0-beta.1.md:23]: i think there are now:
```
- 306 new commits merged
- 4 breaking changes :warning:
- 11 feature additions ✨
- 37 bugs fixed 🐛
```

File: CHANGELOG/v1.7.0-beta.1.md [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+- github.com/golang/mock: [v1.4.4 → v1.4.0](https://github.com/golang/mock/compare/v1.4.4...v1.4.0)
+- github.com/golang/protobuf: [v1.5.3 → v1.5.4](https://github.com/golang/protobuf/compare/v1.5.3...v1.5.4)
+- github.com/google/cel-go: [v0.16.1 → v0.17.7](https://github.com/google/cel-go/compare/v0.16.1...v0.17.7)
+- github.com/google/uuid: [v1.3.1 → v1.4.0](https://github.com/google/uuid/compare/v1.3.1...v1.4.0)
+- github.com/googleapis/enterprise-certificate-proxy: [v0.3.1 → v0.3.2](https://github.com/googleapis/enterprise-certificate-proxy/compare/v0.3.1...v0.3.2)
+- github.com/googleapis/google-cloud-go-testing: [bcd43fb → 1c9a4c6](https://github.com/googleapis/google-cloud-go-testing/compare/bcd43fb...1c9a4c6)
+- github.com/gorilla/websocket: [v1.4.2 → v1.5.0](https://github.com/gorilla/websocket/compare/v1.4.2...v1.5.0)
+- github.com/nats-io/nats.go: [v1.30.2 → v1.31.0](https://github.com/nats-io/nats.go/compare/v1.30.2...v1.31.0)
+- github.com/nats-io/nkeys: [v0.4.5 → v0.4.6](https://github.com/nats-io/nkeys/compare/v0.4.5...v0.4.6)
+- github.com/onsi/ginkgo/v2: [v2.13.1 → v2.17.0](https://github.com/onsi/ginkgo/compare/v2.13.1...v2.17.0)
```
Comment (cahillsf) [inline CHANGELOG/v1.7.0-beta.1.md:350]:
```suggestion
- github.com/onsi/ginkgo/v2: [v2.13.1 → v2.17.1](https://github.com/onsi/ginkgo/compare/v2.13.1...v2.17.1)
```

File: CHANGELOG/v1.7.0-beta.1.md [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+- e2e: Test/framework: expose CopyAndAmendClusterctlConfig function (#10086)
+- e2e: Test: combine Finalizers tests to default quick start tests (#10039)
+- e2e: Use --wait-providers in test framework InitWithBinary func (#10149)
+- e2e: Use manager in test extension (#10097)
+- e2e: Add scale testing for upgrades (#9077)
+- Machine: Be more explicit when skipping node deletion (#10137)
+- Machine: Mark Machine healthy condition as unknown if we can't list wl nodes (#9864)
+- MachineDeployment: Set revision on machinedeployment and patch via defer (#10160)
+- MachinePool: Use SSA Patch to create machines in MP controller (#9791)
+- MachineSet: MS: improve replica defaulting for autoscaler (#9649)
```
Comment (cahillsf) [inline CHANGELOG/v1.7.0-beta.1.md:182]:
```suggestion
- MachineSet: improve replica defaulting for autoscaler (#9649)
```

File: CHANGELOG/v1.7.0-beta.1.md [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+- KCP/MachineSet: KCP/MS remove hardcoded WithStepCounterIf(false) (#9772)
```
Comment (cahillsf) [inline CHANGELOG/v1.7.0-beta.1.md:183]:
```suggestion
- KCP/MachineSet: remove hardcoded WithStepCounterIf(false) (#9772)
```

File: CHANGELOG/v1.7.0-beta.1.md [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+- CI: Bump kubebuilder envtest to 1.29.0 (#10013)
+- CI: DumpResourcesForCluster should fail fast for i/o errors (#10238)
+- CI: Ensure build images contains correct binary and for correct architecture (#9932)
+- CI: Fix Make target generate-go-openapi (#10161)
+- CI: Fix TestPatchNode flake (#10287)
+- CI: Hack/prowjob-gen skip creating empty files (#10022)
+- CI: Hack: implement generator for prowjobs (#9937)
+- CI: Makefile: bump dependencies (#10236)
+- CI: Test/e2e: bump WorkloadKubernetesVersion for v1.6 clusterctl upgrade test (#10017)
+- CI: Test: adjust capd dockerfile so the binary exposes the package in the built binraries path variable (#10030)
```
Comment (nawazkh) [inline CHANGELOG/v1.7.0-beta.1.md:115]: Looks like duplicated area tags in here.

**Pull request merged**